crates/oxc_linter/src/rules/array_callback_return/return_checker.rs

```diff
@@ -0,0 +1,396 @@
+use oxc_ast::ast::{BlockStatement, Statement, SwitchCase};
+
+/// Lattice structure describe the behavior of a statement with respect to
+/// returning from the function
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum StatementReturnLattice {
+    // Always return
+    AlwaysExplicit,
+    AlwaysImplicit,
+    AlwaysMixed,
+
+    // Return on some paths
+    SomeExplicit,
+    SomeImplicit,
+    SomeMixed,
+
+    // Return on no path
+    NotReturn,
+}
```

Comment:
Lattice is a name I borrowed from Sparse Conditional Constant Propagation, where each variable's state(lattice) transitions between `MaybeConstant, Constant, NotConstant`

Reply:
I think it is some structure in mathematics but I don't know exactly...

Reply:
🤯

Reply:
Can you also add the comments about the name? We want the next to learn more about this, any link to any resources will be even better.

Reply:
I found out it's not a "lattice" according to the definition. Will make another name and add some comments on my reasoning.